src/linalg/impl_linalg.rs

```diff
@@ -578,39 +578,37 @@ pub fn general_mat_vec_mul<A, S1, S2, S3>(alpha: A,
         #[cfg(feature = "blas")]
         macro_rules! gemv {
             ($ty:ty, $gemv:ident) => {
-                if blas_row_major_2d::<$ty, _>(&a)
-                    && blas_compat_1d::<$ty, _>(&x)
-                    && blas_compat_1d::<$ty, _>(&y)
-                {
-                    let mut a_trans = CblasNoTrans;
-                    let mut a = a.view();
-                    let a_s0 = a.strides()[0];
-                    if a_s0 == 1 && m == k {
-                        a = a.reversed_axes();
-                        a_trans = CblasTrans;
-                    }
-                    // adjust strides, these may [1, 1] for column matrices
-                    let a_stride = cmp::max(a.strides()[0] as blas_index, k as blas_index);
-                    let x_stride = x.strides()[0] as blas_index;
-                    let y_stride = y.strides()[0] as blas_index;
-
-                    unsafe {
-                        blas_sys::$gemv(
-                        CblasRowMajor,
-                        a_trans,
-                        m as blas_index, // m, rows of Op(a)
-                        k as blas_index, // n, cols of Op(a)
-                        cast_as(&alpha),     // alpha
-                        a.ptr as *const _,   // a
-                        a_stride, // lda
-                        x.ptr as *const _,   // x
-                        x_stride,
-                        cast_as(&beta),      // beta
-                        y.ptr as *mut _,     // x
-                        y_stride,
-                    );
+                if let Some(layout) = blas_layout::<$ty, _>(&a) {
+                    if blas_compat_1d::<$ty, _>(&x)
+                        && blas_compat_1d::<$ty, _>(&y)
+                    {
+                        let a_trans = CblasNoTrans;
+                        let a_stride = match layout {
+                            CBLAS_LAYOUT::CblasRowMajor => a.strides()[0] as blas_index,
+                            CBLAS_LAYOUT::CblasColMajor => a.strides()[1] as blas_index,
+                        };
+
+                        let x_stride = x.strides()[0] as blas_index;
+                        let y_stride = y.strides()[0] as blas_index;
+
+                        unsafe {
+                            blas_sys::$gemv(
+                                layout,
+                                a_trans,
+                                m as blas_index, // m, rows of Op(a)
+                                k as blas_index, // n, cols of Op(a)
+                                cast_as(&alpha),     // alpha
+                                a.ptr as *const _,   // a
+                                a_stride, // lda
+                                x.ptr as *const _,   // x
+                                x_stride,
+                                cast_as(&beta),      // beta
+                                y.ptr as *mut _,     // x
+                                y_stride,
```

Comment:
Important note, the old and new code look wrong for the strided 1d vectors here? I'm thinking of how blas requires pointers to be passed when stride < 0.